src/boot/bootctl-uki.c

```diff
@@ -0,0 +1,111 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "bootctl.h"
+#include "bootctl-uki.h"
+#include "fd-util.h"
+#include "parse-util.h"
+#include "pe-header.h"
+
+#define MAX_SECTIONS 96
+
+static const uint8_t dos_file_magic[2] = "MZ";
+static const uint8_t pe_file_magic[4] = "PE\0\0";
+
+static const uint8_t name_osrel[8] = ".osrel";
+static const uint8_t name_linux[8] = ".linux";
+static const uint8_t name_initrd[8] = ".initrd";
+
+static int pe_sections(FILE *uki, struct PeSectionHeader **ret, uint16_t *ret_n) {
```

Comment:
doesn't matter, but I'd probably change the type of `ret_n` to `size_t*`. i.e. use the native type we measure memory sizes in. the fact that in PE this is a 16bit entity is kinda an implementation detail i would say and should not matter once the stuff is loaded into memory.

but also, doesn't really matter